### ISSUE TITLE
Question regarding setup

### ISSUE THREAD

**Issue by mhartington** [Jaykul/PowerLine#10]
Hi there!

I just installed and have been going over the readme and had a question.

Currently on my mac/linux machines, I have my powerline prompt setup to this.

 
<img width="436" alt="screen shot 2017-01-04 at 1 30 40 pm" src="https://cloud.githubusercontent.com/assets/2835826/21654295/f015f260-d282-11e6-84a9-06acf521c794.png">

Where instead of just printing out the PWD, it prints the whole path from $HOME. But if the path is more than 3 directories deep, it shortens is with the `...`.

Would such a setup be possible?

**Comment by Jaykul** [comment]
Hey @mhartington sorry I didn't reply sooner (I really need to shorten the list of projects I get emails about to just mine, so I don't miss issues).

Yes, you can do that, but I hadn't built a slick function for it yet:

```posh
function Get-PowerLinePath { 
param($Length = 5)
    $dir = $pwd.Path;
    $buffer = @()
    while($dir) {
        $buffer += @{ Object = (Split-Path $dir -Leaf) }
        $dir = Split-Path $dir
        if(!($Length -= 1)) {
            $buffer += @{ Object = "…"; }
            break
        }
    }
    [Array]::Reverse($buffer)
    $buffer
}
```

You need to define that, and add _a call to that function_ into your PowerLinePrompt. I'd paste a screenshot, but I don't have a PowerLine font on this PC, so same-color separators look dumb 😊 

I'm going to leave this open until I integrate that function to the module

**Comment by Jaykul** [comment]
Ok, I wrote a much better version of this in [/Source/Public/Get-SegmentedPath.ps1](https://github.com/Jaykul/PowerLine/blob/master/Source/Public/Get-SegmentedPath.ps1)

**Comment by Jaykul** [comment]
This needs a warning at import time, as a minimum, I think. Especially since those of us with production code and CI may want to use more than one version at various times (i.e. to test whether we can upgrade the build system)

**Comment by Thraka** [comment]
@Jaykul I'm having a hard time figuring out how to customize the > that gets placed with `Get-SegmentedPath`  when it is in the $prompt variable. Is a setting somewhere?

**Comment by Jaykul** [comment]
@Thraka yes (you should have opened a new issue to ask this, though).

Yes, there is a ColorSeparator and a Separator (and reverse versions of each for right-aligned blocks). You can set them in a hashtable which currently lives as a static property. Type this to see the list, then set the one you want:

```posh
[PoshCode.Pansies.Entities]::ExtendedCharacters
```

**Comment by Thraka** [comment]
@Jaykul I know sorry i broke issue etiquette.. sorry :(

I think I found my fix. The problem is that `ExtendedChars` are characters, not strings. And the delimiter used throughout the scripts (the fancy `>`) burdens my eyes when they jam up close to other text, like in the directory path.

Playing around and learning how this all works, which has been hours of fun, (really it has, this is cool stuff) I figured out how to get what I want. In the `$prompt` array I'm adding spaces around the separator character to override segmentedpath's default.

```powershell
{ New-PromptText { (Get-SegmentedPath -SegmentLimit 7) -join (" $([PoshCode.Pansies.Entities]::ExtendedCharacters["Separator"]) ") } -BackgroundColor DarkBlue -ForegroundColor White },
```
